packages/opencensus-exporter-stackdriver/src/stackdriver-stats-utils.ts

```diff
@@ -0,0 +1,288 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {BucketOptions, DistributionBucket, DistributionValue, LabelKey, LabelValue, Metric, MetricDescriptor as OCMetricDescriptor, MetricDescriptorType, TimeSeriesPoint, Timestamp} from '@opencensus/core';
+import * as os from 'os';
+import * as path from 'path';
+import {Distribution, LabelDescriptor, MetricDescriptor, MetricKind, Point, TimeSeries, ValueType} from './types';
+
+/**
+ * Util methods to convert OpenCensus Metrics data models to StackDriver
+ * monitoring data models.
+ */
+export class StackdriverStatsExporterUtils {
+  static readonly OPENCENSUS_TASK: string = 'opencensus_task';
+  static readonly OPENCENSUS_TASK_DESCRIPTION: string =
+      'Opencensus task identifier';
+  static readonly OPENCENSUS_TASK_VALUE_DEFAULT =
+      StackdriverStatsExporterUtils.generateDefaultTaskValue();
+
+  /**
+   * Returns a task label value in the format of 'nodejs-<pid>@<hostname>'.
+   */
+  static generateDefaultTaskValue(): string {
+    const pid = process.pid;
+    const hostname = os.hostname() || 'localhost';
+    return 'nodejs-' + pid + '@' + hostname;
+  }
+
+  /**
+   * Converts a OpenCensus MetricDescriptor to a StackDriver MetricDescriptor.
+   * @param metricDescriptor The OpenCensus MetricDescriptor.
+   */
+  static createMetricDescriptorData(
+      metricDescriptor: OCMetricDescriptor, metricPrefix: string,
+      displayNamePrefix: string): MetricDescriptor {
+    return {
+      type: this.getMetricType(metricDescriptor.name, metricPrefix),
+      description: metricDescriptor.description,
+      displayName:
+          this.createDisplayName(metricDescriptor.name, displayNamePrefix),
+      metricKind: this.createMetricKind(metricDescriptor.type),
+      valueType: this.createValueType(metricDescriptor.type),
+      unit: metricDescriptor.unit,
+      labels: this.createLabelDescriptor(metricDescriptor.labelKeys)
+    } as MetricDescriptor;
+  }
+
+  /**
+   * Converts metric's timeseries to a list of TimeSeries, so that metric can be
+   * uploaded to StackDriver.
+   * @param metric The metric to get TimeSeries information from.
+   */
+  static createTimeSeriesList(
+      metric: Metric,
+      monitoredResource: {type: string, labels: {[key: string]: string}},
+      metricPrefix: string): TimeSeries[] {
+    const timeSeriesList: TimeSeries[] = [];
+
+    // TODO(mayurkale): Use Resource API here, once available (PR#173)
+    const metricDescriptor = metric.descriptor;
+    const metricKind = this.createMetricKind(metricDescriptor.type);
+    const valueType = this.createValueType(metricDescriptor.type);
+
+    for (const timeSeries of metric.timeseries) {
+      timeSeriesList.push({
+        metric: this.createMetric(
+            metricDescriptor, timeSeries.labelValues, metricPrefix),
+        resource: monitoredResource,
```

Comment:
Just to double check: it's fine to use the same `monitoredResource` here (i.e no need to convert)?

Reply:
Currently no, but once we have monitored resource util in place (#173) we need to do the conversion same as Java's implementation.